src/frontend/src/components/FillForms/UserDataForm.tsx

```diff
@@ -1,337 +1,294 @@
+};
 
-  private buildDob = () => {
-    if (this.props.aliases.length > 0) {
-      const firstAlias = this.props.aliases[0];
-      return firstAlias.birth_date;
-    } else {
-      return "";
-    }
-  };
+const isZipCode = (str: string) => {
+  //the zipCode RegEx accepts any 5 digit entry ex. "12345"
+  return /[0-9][0-9][0-9][0-9][0-9]/.test(str); // Check for empty string included
+};
 
-  public state: State = {
-    name: this.buildName(),
-    dob: this.buildDob(),
-    mailingAddress: "",
-    phoneNumber: "",
-    city: "",
-    state: "",
-    zipCode: "",
-    invalidZipCode: false,
-    invalidPhone: false,
-    invalidBirthDate: false,
-  };
+export default function UserDataForm() {
+  const aliases = useAppSelector(selectSearchFormValues).aliases;
+  const dispatch = useDispatch();
+  const [name, setName] = useState(buildName());
+  const [dob, setDob] = useState(aliases[0].birth_date);
+  const [mailingAddress, setMailingAddress] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [city, setCity] = useState("");
+  const [state, setState] = useState("Oregon");
+  const [zipCode, setZipCode] = useState("");
+  const [errorMessages, setErrorMessages] = useState<string[]>([]);
+  const loadingExpungementPacket = useAppSelector(
+    (state) => state.search.loadingExpungementPacket
+  );
 
-  phoneNumberPattern = new RegExp(".*[0-9].*");
-  zipCodePattern = new RegExp("[0-9][0-9][0-9][0-9][0-9].*");
+  const [showModal, setShowModal] = useState(false);
 
-  public validateForm = () => {
-    const phoneNumberMatch = this.phoneNumberPattern.exec(
-      this.state.phoneNumber
-    );
-    const zipCodeMatch = this.zipCodePattern.exec(this.state.zipCode);
-    return new Promise((resolve: (value?: unknown) => void) => {
-      this.setState<any>(
-        {
-          invalidZipCode: this.state.zipCode.length > 0 && !zipCodeMatch,
-          invalidPhone: this.state.phoneNumber.length > 0 && !phoneNumberMatch,
-          invalidBirthDate:
-            this.state.dob.length > 0 &&
-            moment(this.state.dob, "M/D/YYYY", true).isValid() === false,
-        },
+  function buildName() {
+    const { first_name, middle_name, last_name } = aliases[0];
+    let fullName = "";
+    fullName += first_name ? first_name + " " : "";
+    fullName += middle_name ? middle_name + " " : "";
+    fullName += last_name;
+
+    return fullName.includes("*") ? "" : fullName || " ";
+  }
+
+  function allInputsValid() {
+    const errorMessages = {
+      "Zip code must lead with five digits":
+        isPresent(zipCode) && !isZipCode(zipCode),
+      "Phone number must be a valid 10 digit number":
+        isPresent(phoneNumber) && !isPhoneNumber(phoneNumber),
+    };
 
-        resolve
-      );
-    });
-  };
+    if (errorMessagesObjectToArray(errorMessages).length === 0) return true;
+
+    setErrorMessages(errorMessagesObjectToArray(errorMessages));
```

Comment:
The function name `allInputsValid` suggests to me that there are no "side-effects" in the function - just a simple evaluation of variables which resolve to true/false. I would prefer a name that suggests the side-effect. Or figure out how to move the side-effect out (while still keeping things simple).

Reply:
would something like `inputsMeetCriteria` work?

Reply:
Hmm I don't think that implies a side-effect. Maybe inputsMeetCriteriaAndSetErrors . Kind of a mouthful but at least it is clear.

Reply:
I have updated that function name along with completing the other recommendations you made above